src/dune_rules/ml_sources.ml

```diff
@@ -42,6 +44,18 @@ module Modules = struct
               "Executable %S appears for the second time in this directory" name
           ]
     in
+    let melange_emits =
+      match
+        String.Map.of_list_map melange_emits ~f:(fun (target, m, obj_dir) ->
+            (snd target, (m, obj_dir)))
+      with
+      | Ok x -> x
+      | Error (name, _, (target, _, _)) ->
+        User_error.raise ~loc:(fst target)
+          [ Pp.textf "Target %S appears for the second time in this directory"
+              name
+          ]
+    in
```

Comment:
You'll need to modify the reverse map right below as well.

Reply:
I saw it, but i'm not sure how. `rev_map` is of type `Buildable.t Module_name.Map.t`.

Everything suggests we need a "thinner" version of `Buildable.t`?

Reply:
"`Modulesable.t`" 😆  or `With_modules_field.t` (although in melange it'd be called `entries`...)

Reply:
The appropriate type would be:

```
[ `Buildable of Buildable.t | `Melange of Melange_stanza.Emit.t ]
```

Reply:
I tried something in https://github.com/ocaml/dune/pull/6349/commits/398e0eadd595374362e950fc44ca0fbdff493a95. Lmk if it's what you had in mind, ended up using regular variant as there's not much difference and things are cleaner.